### ISSUE TITLE
Prettier config - at  Setting up Prettier and VS Code

### ISSUE THREAD

**Issue by MarekOk** [jonasschmedtmann/complete-javascript-course#288]
Hello, 
 As for September '22 this - two years old version of the course have obviously many differences ( at least in interface of Visual Studio Code )  and suggested by Jonas configuration, as may be not crucial - can be significant along the remaining 50 hours (hundreds actually ) of following the guidelines. 
  It may be also difference between MAC ( which Author is using ) and WIN OS ( which I'm using WIN 10 Pro ) - or simply new VSC version. 
   By following the video about setting the ''Prettier and VS Code''  - I'm unable to change single quotes ( ' ' ) instead auto correcting double quotes ( "" ) while writing the code.
    First of all - I can't create New File in VSC by clicking the right mouse button - as the Author in the video.
    Once I'm creating New File - I can't change the name of the file, which is still as ''untitled''. Once I save the file ( as  .txt ) I can change the name, but still - it doesn't help with the outcome by writing ''the object'' ( command? ) to direct Prettier app not to changing single quotes in to double quotes... 
 Exactly the same thing with changing default function of Prettier - putting parentheses into a single parameter ( which Jonas ''don't like'' and not recommending... 
The main issue may be, that I can create only new .txt file, which doesn't help with anything... 
 Anyone had similar issues? 
  Thank you for any hints or/and help... 
   Mark
![__prettier single quote](https://user-images.githubusercontent.com/114067876/191482068-35635d79-943a-4adf-8ece-e4e5988aa6d3.jpg)

**Comment by akozdev** [comment]
Hi Mark

The config file for Prettier should be named as **.prettierrc.json**. From what I see in your editor, it's called **.prettier.rc.txt**.

This file contains JSON, which supports only quotation marks (apostrophes are not supported). This code should work:
```json
{
  "singleQuote": true,
  "arrowParens": "avoid"
}
```
To be able to create new files from the sidebar, you need to open a directory (that contains your project) in VS Code. To do so in VS Code: **File > Open Folder... > choose your project directory**
<img width="575" alt="vscode_file" src="https://user-images.githubusercontent.com/51536161/191552644-358dbde7-465d-4cd2-a843-f9117c685a84.png">
After that, you should be able to Right-click on the sidebar and create a new file.
Let me know if you were able to create the .prettierrc.json file.

**Comment by MarekOk** [comment]
![node02 live-server Jonas](https://user-images.githubusercontent.com/114067876/191566516-65f97236-4a03-4231-973d-c16618e10d99.jpg)
![node01 Jonas](https://user-images.githubusercontent.com/114067876/191566512-499a344a-46f3-47ba-8cf7-e20ae50d7b9a.jpg)
    
![node01 mine](https://user-images.githubusercontent.com/114067876/191566460-7ad2acc8-0eae-4fe7-b935-3c82b2829179.jpg)
![node02 live-server mine](https://user-images.githubusercontent.com/114067876/191566465-259624f4-2c6d-4b77-b1bd-cc616f88c7ae.jpg)

**Comment by MarekOk** [comment]
OMG - Thank you Aleksander - it worked like a charm (!!) - thank you very much for clear explanation. 
 It works flawlessly now. 
  And I'm actually stuck on another (lame) setting - maybe you'd be able to give me a hint in this matter as well... 
  I'm unable to syncro my  freshly installed Node.js with VSCode as for live-server :( 
  Most likely this is because of - that I don't know how to chose directory/folder in terminal ( for the one which contains the project ) , but may be a different issue... I did re-installed Node.js itself as also VSCode rebooting multiple times... 
  I'm currently follow a JS online course, so completely newbie in those programs/apps 
  Two first SS are from mine terminal and last two SS from the Author terminal... 
   Dziękuję jak nie wiem co za dotychczasową pomoc/radę :)
    Serdecznie, 
    Marek 
    PS. No idea why this text wasn't attached to the response...

**Comment by akozdev** [comment]
No problem 🙂 It's completely normal to have problems with these things as a beginner. You will get used to it with time and practice. Programming in general requires a lot of patience because things like to break for different reasons.

For the live-server, you will need two things:
1) PowerShell (a program that runs commands in your Terminal) doesn't allow to run scripts (like live-server) by default for security reasons. You can however enable running scripts with the command below
```powershell
Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
```
You need to run the above command from your Terminal with administrator rights (you can open VS Code as administrator (search for VS Code in your OS > Right-click > Run as administrator)).

Note, if you ever want to go back to the default execution policy in PowerShell, run this command from Terminal with administrator rights
```powershell
Set-ExecutionPolicy -ExecutionPolicy Default -Scope CurrentUser
```
You can also read more about execution policies in PowerShell at:
[https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-7.2](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-7.2)

2) You will need to navigate to your project directory in Terminal to run live-server from there. This is done with the `cd` command, which means "change directory". I don't know where exactly your project is, but let's assume you want to first go to the _complete-javascript-course_ directory that exists in your user's directory. Let's say you're currently in the _C:/Users/Dell_; to navigate to co the _complete-javascript-course_ directory, run `cd complete-javascript-course` from your Terminal.  
You can also use absolute paths, for example, let's say that the _complete-javascript-course_ directory is at _C:/Users/Dell/complete-javascript-course_. To navigate there, you can just write `cd C:/Users/Dell/complete-javascript-course`.

Let me know if it works for you.

**Comment by akozdev** [comment]
Also, there is an easier way to run a development server (I think it was mentioned in the course). Instead of using the live-server package, you can install the [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) VS Code extension made by Ritwick Dey.

Then, you just need to open your project in VS Code (as in previous post), and click on the 'Go Live' button from the bottom status bar
<img width="450" alt="go_live" src="https://user-images.githubusercontent.com/51536161/191578435-1b25431e-7b24-4c09-b02d-1b46941f3152.png">

**Comment by MarekOk** [comment]
Yes - as I followed the course video and installed ( also multiple times :D ) the VSCode app Live Server - but for... ''some reason'' ( :D ) I could not connect it to the console, in spite of spending couple of hours on ytb and rebooting, uninstalling and chosing older version of this app... 
 Funny, that after fixing ( thanks to YOU :) 👍 NodeJS issue - Live Server app also connected without problems... 
  I'm highly impressed by the professional approach and your skill in fixing problems and willingness to help 🙏🏻🙏🏻🙏🏻🤝🏻🤝🏻🤝🏻
Once more - many thanks. I can move forward :)  

![Node READY](https://user-images.githubusercontent.com/114067876/191600187-7ceceb1b-f805-45a8-8083-470c3e6a3a79.jpg)

**Comment by akozdev** [comment]
Glad it works for you 🙂
